packages/lesswrong/server/resolvers/userResolvers.ts

```diff
@@ -22,3 +24,53 @@ addFieldsDict(Users, {
     })
   },
 });
+
+addGraphQLSchema(`
+  type NewUserCompletedProfile {
+    username: String,
+    slug: String,
+    displayName: String,
+    subscribedToDigest: Boolean,
+    usernameUnset: Boolean
+  }
+`)
+
+type NewUserUpdates = {
+  username: string
+  subscribeToDigest: boolean
+}
+
+addGraphQLResolvers({
+  Mutation: {
+    async NewUserCompleteProfile(root: void, { username, subscribeToDigest }: NewUserUpdates, context: ResolverContext) {
+      const { currentUser } = context
+      if (!currentUser) {
+        throw new Error('Cannot change username without being logged in')
+      }
+      // Only for new users. Existing users should need to contact support to
+      // change their usernames
+      if (!currentUser.usernameUnset) {
+        throw new Error('Only new users can set their username this way')
+      }
+      const updatedUser = (await updateMutator({
+        collection: Users,
+        documentId: currentUser._id,
+        set: {
+          usernameUnset: false,
+          username,
```

Comment:
There should be a uniqueness check for username here. Attempting to claim a duplicate username still won't work (because there's a `unique` index on the username field in mongodb), but the failure will be an inelegant one.